src/main/java/gregtech/common/terminal/app/prospector/widget/WidgetProspectingMap.java

```diff
@@ -213,25 +228,111 @@ public void drawInForeground(int mouseX, int mouseY) {
+                                    if (oreInfo.get(name) > maxAmount[0]) {
+                                        maxAmount[0] = oreInfo.get(name);
+                                        color = OreDictUnifier.getMaterial(OreDictUnifier.get(dict)).material.getMaterialRGB();
+                                    }
                                 }
                             });
                         }
                     }
                 }
-                oreInfo.forEach((name, count)->tooltips.add(name + " --- " + count));
-            } else if(this.mode == 1){
+                oreInfo.forEach((name, count) -> {
+                    tooltips.add(name + " --- " + count);
+                    hoveredNames.add(name);
+                });
+            } else if (this.mode == 1) {
                 tooltips.add(I18n.format("terminal.prospector.fluid"));
                 if (texture.map[cX][cZ] != null && !texture.map[cX][cZ].isEmpty()) {
                     if (texture.getSelected().equals("[all]") || texture.getSelected().equals(texture.map[cX][cZ].get((byte) 1))) {
+                        FluidStack fluidStack = FluidRegistry.getFluidStack(texture.map[cX][cZ].get((byte) 1), 1);
                         tooltips.add(I18n.format("terminal.prospector.fluid.info",
-                                FluidRegistry.getFluidStack(texture.map[cX][cZ].get((byte) 1),1).getLocalizedName(),
+                                        fluidStack.getLocalizedName(),
                                 texture.map[cX][cZ].get((byte) 2),
                                 texture.map[cX][cZ].get((byte) 3)));
+                        hoveredNames.add(fluidStack.getLocalizedName());
+                        int amount = Integer.parseInt(texture.map[cX][cZ].get((byte) 2));
+                        if ( amount > maxAmount[0]) {
+                            maxAmount[0] = amount;
+                            color = fluidStack.getFluid().getColor(fluidStack);
+                        }
                     }
                 }
             }
             this.drawHoveringText(ItemStack.EMPTY, tooltips, 300, mouseX, mouseY);
             GlStateManager.color(1.0F, 1.0F, 1.0F);
         }
     }
+
+    @Override
+    public boolean mouseClicked(int mouseX, int mouseY, int button) {
+
+        int cX = (mouseX - this.getPosition().x) / 16;
+        int cZ = (mouseY - this.getPosition().y) / 16;
+
+        if (cX >= chunkRadius * 2 - 1 || cZ >= chunkRadius * 2 - 1)
+            return false;
+
+        int xDiff = cX - (chunkRadius - 1);
+        int zDiff = cZ - (chunkRadius - 1);
+
+        int xPos = ((Minecraft.getMinecraft().player.chunkCoordX + xDiff) << 4) + 8;
+        int zPos = ((Minecraft.getMinecraft().player.chunkCoordZ + zDiff) << 4) + 8;
+
+        BlockPos b = new BlockPos(xPos, Minecraft.getMinecraft().world.getHeight(xPos, zPos), zPos);
+        if (System.currentTimeMillis() - lastClicked < 200) {
```

Comment:
You may want to add description in the prospector gui to tell player double click to add waypoint